### PR TITLE
Add preflight package

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Preflight package implements should implement
all preflight functionality per one Module.
Current commit will only implement image verification:
image presence and correct kernel version

MGMT-11438

### CODE REVIEW

File: internal/module/kernelmapper.go [hunk 1]
```diff
@@ -19,14 +19,14 @@ const (
...
-	kernelRHELRelease  string `subst:"KERNEL_RHEL_RELEASE"`
-	kernelRHELArch     string `subst:"KERNEL_RHEL_ARCH"`
+	KernelFullVersion  string `subst:"KERNEL_FULL_VERSION"`
+	KernelVersionMMP   string `subst:"KERNEL_XYZ"`
+	KernelVersionMajor string `subst:"KERNEL_X"`
+	KernelVersionMinor string `subst:"KERNEL_Y"`
+	KernelVersionPatch string `subst:"KERNEL_Z"`
+	KernelRHELVersion  string `subst:"KERNEL_RHEL_VERSION"`
+	KernelRHELRelease  string `subst:"KERNEL_RHEL_RELEASE"`
+	KernelRHELArch     string `subst:"KERNEL_RHEL_ARCH"`
```
Comment (qbarrand) [inline internal/module/kernelmapper.go:29]: How about we drop those fields and keep them for downstream only? We are about to push this code into Kubernetes.

File: internal/preflight/preflight.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	osConfig := module.NodeOSConfig{KernelFullVersion: kernelVersion}
+	mapping, err = p.kernelAPI.PrepareKernelMapping(mapping, &osConfig)
+	if err != nil {
+		return false, fmt.Sprintf("Failed to substitute template in kernel mapping in the module %s for kernel version %s", mod.Name, kernelVersion)
+	}
+
+	return p.verifyImage(ctx, mapping, mod, kernelVersion)
+}
+
+func (p *preflight) verifyImage(ctx context.Context, mapping *kmmv1beta1.KernelMapping, mod *kmmv1beta1.Module, kernelVersion string) (bool, string) {
```
Comment (qbarrand) [inline internal/preflight/preflight.go:62]: Could this just return an error?

File: internal/preflight/preflight.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	}
+}
+
+type preflight struct {
+	client      client.Client
+	registryAPI registry.Registry
+	kernelAPI   module.KernelMapper
+}
+
+func (p *preflight) PreflightUpgradeCheck(ctx context.Context, mod *kmmv1beta1.Module, kernelVersion string) (bool, string) {
```
Comment (qbarrand) [inline internal/preflight/preflight.go:47]: Could this just return an error?

File: internal/preflight/preflight.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply internal/preflight/preflight.go:62]: error won't help us. This should return whether the image is verified, and also the message that describes the reason for verification or not verification

File: internal/preflight/preflight.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply internal/preflight/preflight.go:47]: same comment as above. error wont' tell us much, we need the status and the reason message for the status

File: internal/preflight/preflight.go (hunk 2 above)
Comment (qbarrand) [reply internal/preflight/preflight.go:62]: We could do:
- image verified: depends on whether `err` is nil or not
- reason: the error string
WDYT?

File: internal/module/kernelmapper.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply internal/module/kernelmapper.go]: fixed

File: internal/preflight/preflight.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply internal/preflight/preflight.go:62]: I would prefer not to: error means that something went wrong with the flow, while in preflight case there is nothing wrong with the flow, the function  just returns the status of verification.

**Review by qbarrand — APPROVED** [review]

**Pull request merged**